controllers/controllers.go

```diff
@@ -281,8 +281,10 @@ func markNodeAsUpgrading(ctx context.Context, c client.Client, currentNode *core
 		}
 	}
 	if len(upgradingNodes.Items) >= MaxParallelUpgrades {
-		return fmt.Errorf("cannot mark node %s as upgrading, maximum number of parallel upgrading nodes reached (%d)",
-			currentNode.Name, MaxParallelUpgrades)
+		return &UpgradeLimitExceededError{
+			NodeName: currentNode.Name,
+			Count:    len(upgradingNodes.Items),
+		}
```

Comment:
The original error provided the maximum limit value (MaxParallelUpgrades), but the new error structure doesn't capture this information. This could make troubleshooting more difficult. Consider adding it back.